src/pymodaq/utils/tcp_ip/serializer.py

```diff
@@ -617,23 +640,36 @@ def ndarray_deserialization(self) -> np.ndarray:
         ndarray = np.atleast_1d(ndarray)  # remove singleton dimensions
         return ndarray
 
-    def object_deserialization(self):
+    def type_and_object_deserialization(self):
+        """ Deserialize specific objects from their binary representation.
```

Comment:
I think it is good to these methods are invers functions, for example:

```suggestion
        """ Deserialize specific objects from their binary representation (inverse of `Serializer.type_and_object_serialization`).
```

Obviously a similar statement could be above and a note in the documentation, that these functions are the entry/exit points of serialization.